go/vt/vtgate/executor.go

```diff
@@ -1620,7 +1627,13 @@ func (e *Executor) prepare(ctx context.Context, safeSession *SafeSession, sql st
 func (e *Executor) handlePrepare(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, logStats *LogStats) ([]*querypb.Field, error) {
 	// V3 mode.
 	query, comments := sqlparser.SplitMarginComments(sql)
-	vcursor, _ := newVCursorImpl(ctx, safeSession, comments, e, logStats, e.vm, e.VSchema(), e.resolver.resolver)
+	stmt, err := sqlparser.Parse(query)
```

Comment:
I'm a little concerned about the number of places we are adding calls to `sqlparser.Parse`. `getPlan` already does the parsing. Can the ignoreMaxMemoryRows handling be moved there? so you would not set it on the vcursor during initialization, you would set if afterwards.

Reply:
@systay @harshit-gangal WDYT?

Reply:
`handlePrepare`, `handleShow` and `handleSet` are all called from the [old executor code](https://github.com/vitessio/vitess/blob/master/go/vt/vtgate/executor.go#L183), and that means that the parsing has not happened yet. For these three, it's OK to parse inside the `handleXYZ` code.

Reply:
I don't see `handlePrepare` being called from the legacy executor. I can imagine the memory limit being irrelevant for `set` but important for `show`. @spark4 is it posssible at all for you to move this into `getPlan` and still support it for `show`?

Reply:
✅ I updated `getPlan()` to update the vcursor to avoid the extraneous `sqlparser.Parse()` calls